### ISSUE TITLE
Restricted Optional methods?

### ISSUE THREAD

**Issue by m92y** [nus-cs2030/2122-s2#445]
Codecrunch says
`Use of restricted Optional methods`
but i don't know why it says that when the files submitted used only `Optional.ofNullable` and `Optional.empty`

**Comment by m92y** [comment]
no :') thats why im so confused

**Comment by hiimss** [comment]
If I am not wrong the familiar ones that got restricted are ```isPresent()```, ```isEmpty()``` and ```get()```, I am not sure of the other two mentioned above but maybe you can try to comment the relevant code out and submit (although you wont pass the test cases), that is how I got around in finding which methods are being restricted because of the vague specifications

**Comment by celine-chung** [comment]
that's really strange... i used of() and empty() in my implementation and it worked just fine so i think you should probably keep searching to see if there was any additional mention of optional which you overlooked or try using of() instead of ofNullable()? jiayous

**Comment by Zarilex09** [comment]
i think the restricted ones are only isPresent(), isEmpty() and get(), can try to double check, its quite easy to miss out when your code is too long

**Comment by chorhanchean** [comment]
I think ofNullable() is allowed, but personally I used of() and its possible. I think you might have missed out on .get() somewhere in the long lines of code

**Comment by raynang** [comment]
The restricted ones should be isPresent(), isEmpty() and get(). I used .empty() and .of() in my implementations and there were no errors so those should be okay for you too. Maybe you might want to check if you accidentally used .get() anywhere, especially when you use .get() to get service time or rest time, make sure you're using .get() on the supplier and not an optional.

**Comment by denshem10** [comment]
Yes, I encountered the same issue too! I tried to use get() and CodeCrunch blocked me from using it.
As prof mentioned in his lecture, using Optional.get() defeats the purpose of using Optional objects.
`Optional<Event> emptyEvent = Optional.empty()
emptyEvent.get() //this will result in NoSuchElementException error being thrown`

To address your problem, I suggest you use your IDE's search function and look for any Optional getters or other Optional-related methods that are restricted by CodeCrunch. 

How I get an Optional's value is by using the Optional::orElse() method:

`PQ<Event> listOfEvents = new PQ<>(List.of(new Event.....)) //create a PQ of events
Optional<Event> eventFromPQ = pq.polll().first()
Event eventValue = eventFromPQ.orElse(new Event(...)) //if  no value is present, a dummy value will be returned
`
